sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -102,7 +136,12 @@ namespace Azure { namespace Core {
      *
      * @return An HTTP #Azure::Core::Http::RawResponse returned from the service.
      */
-    std::unique_ptr<Http::RawResponse> Poll(Context& context) { return PollInternal(context); }
+    Http::RawResponse const& Poll(Context& context)
+    {
+      context.ThrowIfCancelled();
+      m_rawResponse = PollInternal(context);
+      return *m_rawResponse;
+    }
```

Comment:
This should return Poll() instead of doing the same as Poll()

Will update after storage tests results